lib/Sema/CSDiag.cpp

```diff
@@ -1876,23 +1879,66 @@ bool CalleeCandidateInfo::diagnoseGenericParameterErrors(Expr *badArgExpr) {
                                 argType, archetypesMap))
     return false;
 
+  auto getDecls =
+      [&](ArchetypeType *archetype) -> Optional<GenericDeclWithParent> {
+    auto *env = archetype->getGenericEnvironment();
+    auto paramType = env->mapTypeOutOfContext(archetype);
+
+    if (auto *GTPT = paramType->getAs<GenericTypeParamType>()) {
+      auto *decl = GTPT->getDecl();
+      auto *dc = decl->getDeclContext();
+      auto *parent = dc->getInnermostDeclarationDeclContext();
+      return Optional<GenericDeclWithParent>({decl, parent});
+    } else if (auto *DMT = paramType->getAs<DependentMemberType>()) {
```

Comment:
This violates LLVM style. There is no point in putting in an else if you are returning here. Just use the early return.